### PR TITLE
Remove partial syntax in DQL

### PR DESCRIPTION
**Pull request by greg0ire**
It is no longer possible to use the `PARTIAL` keyword in a DQL query, or to artificially build an AST with a partial object expression. It is still possible to use the result set mapping API to build partial objects.

### CODE REVIEW

**Comment by greg0ire** [comment]
Forbidding building partial objects will be attempted in a separate PR, it will require a substantial rework of the tests, they seem to heavily rely on that ability. A possible solution might be to build a factory, allowing us to specify just the properties that we care about. Note that building such objects through that API does not trigger any deprecation.

**Comment by greg0ire** [comment]
Converting to draft as I've found more stuff to remove

File: UPGRADE.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # Upgrade to 3.0
 
+## BC BREAK: Partial objects are removed
+
+- The `PARTIAL` keyword no longer exists.
```
Comment (beberlei) [inline UPGRADE.md]:
```suggestion
- The `PARTIAL` keyword in DQL no longer exists.
```

**Review by beberlei — DISMISSED** [review]

**Review by beberlei — APPROVED** [review]

**Comment by walva** [comment]
I wonder what was the reasoning behind this move. I use this feature time to time and I know ApiPlatform rely on it.

**Comment by greg0ire** [comment]
The place to discuss that is https://github.com/doctrine/orm/issues/8471

**Pull request merged**